src/main/java/de/tum/cit/aet/artemis/iris/service/session/IrisCourseChatSessionService.java

```diff
@@ -120,10 +145,62 @@ public void requestAndHandleResponse(IrisCourseChatSession session) {
+    private void requestAndHandleResponse(IrisCourseChatSession irisSession, String variant, CompetencyJol competencyJol) {
+        var session = (IrisCourseChatSession) irisSessionRepository.findByIdWithMessagesAndContents(irisSession.getId());
+
+        var courseId = session.getCourse().getId();
+        var userId = session.getUser().getId();
+
+        var courseDTO = PyrisExtendedCourseDTO.from(loadCourseWithParticipationOfStudent(courseId, userId));
+        var metricsDTO = learningMetricsService.getStudentCourseMetrics(userId, courseId);
+        var competencyJolDTO = Optional.ofNullable(competencyJol).map(CompetencyJolDTO::of).orElse(null);
+        var conversation = session.getMessages().stream().map(PyrisMessageDTO::from).toList();
+        var userDTO = PyrisUserDTO.from(session.getUser());
+
+        // @formatter:off
+        pyrisPipelineService.executePipeline(
+                "course-chat",
+                variant,
+                token -> new CourseChatJob(token, courseId, session.getId()),
+                executionDto -> new PyrisCourseChatPipelineExecutionDTO(
+                        courseDTO,
+                        metricsDTO,
+                        competencyJolDTO,
+                        conversation,
+                        userDTO,
+                        executionDto.settings(),
+                        executionDto.initialStages()
+                ),
+                stages -> irisChatWebsocketService.sendStatusUpdate(session, stages)
+        );
+        // @formatter:on
+    }
+
+    /**
+     * Load the course with the participation of the student and set the participations on the exercises.
+     * <p>
+     * Spring Boot 3 does not support conditional left joins, so we have to load the participations separately.
+     *
+     * @param courseId  the id of the course
+     * @param studentId the id of the student
+     */
+    private Course loadCourseWithParticipationOfStudent(long courseId, long studentId) {
+        Course course = courseRepository.findWithEagerExercisesAndLecturesAndAttachmentsAndLectureUnitsAndCompetenciesAndExamsById(courseId).orElseThrow();
+        List<StudentParticipation> participations = studentParticipationRepository.findByStudentIdAndIndividualExercisesWithEagerSubmissionsResultIgnoreTestRuns(studentId,
+                course.getExercises());
+
+        Map<Long, Set<StudentParticipation>> participationMap = new HashMap<>();
+        for (StudentParticipation participation : participations) {
+            Long exerciseId = participation.getExercise().getId();
+            participationMap.computeIfAbsent(exerciseId, k -> new HashSet<>()).add(participation);
+        }
+
+        course.getExercises().forEach(exercise -> {
+            Set<StudentParticipation> exerciseParticipations = participationMap.getOrDefault(exercise.getId(), Set.of());
+            exercise.setStudentParticipations(exerciseParticipations);
+        });
```

Comment:
```suggestion
        course.getExercises().forEach(exercise -> {
            Set<StudentParticipation> exerciseParticipations = participations.stream().filter(participation -> participation.getExercise().getId().equals(exercise.getId())).collect(Collectors.toSet());
            exercise.setStudentParticipations(exerciseParticipations);
        });
```

Reply:
I agree that this is cleaner code, but I intentionally didn't reimplement anything in this PR, only moved things. So I would prefer to keep it the way it is, that way we can be sure that nothing will break.